app/src/main/java/com/jetpack/compose/learning/maps/overlay/MapGroundOverlayViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package com.jetpack.compose.learning.maps.overlay
+
+import androidx.lifecycle.ViewModel
+import com.jetpack.compose.learning.R
+import com.jetpack.compose.learning.maps.GroundOverlayMapUIState
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class MapGroundOverlayViewModel : ViewModel() {
+    private val imageList = listOf(
+        R.drawable.ic_statue_of_liberty,
+        R.drawable.wp1,
+        R.drawable.wp2,
+        R.drawable.wp3,
+        R.drawable.wp4,
+        R.drawable.wp5,
+        R.drawable.wp6,
+        R.drawable.wp7
+    )
+    private val _state = MutableStateFlow(GroundOverlayMapUIState())
+    val state: StateFlow<GroundOverlayMapUIState>
+        get() = _state
+
+    fun changeImage() {
+        val randomImage = imageList.random()
+        if (randomImage != _state.value.imageResource) {
+            _state.value = _state.value.copy(imageResource = randomImage)
+        } else {
+            changeImage()
```

Comment:
Seems method invocation will be infinite. It is calling itself in the else condition.